### ISSUE TITLE
TCL Power

### ISSUE THREAD

**Issue by Baubascat** [Greg-MM/C4-AndroidTV#5]
I am using your driver with a TCL 98R754. The Power On works but not the Power Off. Also the Conditionals "If Device is on" and "If Device is off" aren't recognized even though the Power Status is showing correct ON and OFF states in the Properties tab. I am currently using KeyCode 26 mapped to STOP to turn off the TV.

**Comment by Greg-MM** [comment]
That should be easy to fix, I was planning on making the ON and OFF KeyCode map-able, the status shouldn't be too hard either,   I'll add that to the next version that I publish

**Comment by Baubascat** [comment]
ON and OFF KeyCode map-able would be good but the current Power ON is working as a discrete ON only code. The 26 code is a Power Toggle code. When you say "That should be easy to fix" do you mean you can get the Power OFF to be a discrete Power Off code and not a Power Toggle code? If so, that would be great!

**Comment by Greg-MM** [comment]
You can try pasting the below code into the LUA tab (will keep code until director restarts), I added the 
_if (Properties["Power Status"] == "ON") then	SendKey(KEYCODE_POWER)	end_
to the OFF command, it is setup to leave the device on currently, I will make it a configurable option as the MiBox that I use does not like to turn back on (even with stock remote, seems like I am not alone either)

```
function ReceivedFromProxy(BindingID, strCommand, tParams)
	DebugHeader("ReceivedFromProxy (" .. BindingID .. "): " .. strCommand)
	if (tParams ~= nil) then	Debug(tParams)	end
	if(Properties["Connection"] == "OFFLINE") then
			Debug("Connection OFFLINE, Attempting To Connect...")
			LUA_ACTION.ConnectToCommand(nil)
	end
	if (BindingID == 5001) then
		if (strCommand == "ON") then
			if (Properties["Send WOL on ON"] == "True") then
				SendMagicPacket()
			end
			if (Properties["Power Status"] == "OFF") then
				SendKey(KEYCODE_POWER)
			end
		elseif (strCommand == "OFF") then
			if (Properties["On Room OFF"] == "Pause") then
				SendKey(tonumber(Properties["PAUSE_ROOM_OFF Mapping"]))
			end
			if (Properties["Power Status"] == "ON") then
				SendKey(KEYCODE_POWER)
			end
		else
			if (ProcessInputCommand(strCommand, tParams) == false) then
				DebugDivider("!")
				Debug("COMMAND UNKNOWN: "..strCommand)
				DebugDivider("!")
			end
		end	
elseif (BindingID==5002) then -- Mini Driver Commands
		--TODO: Implement Mini Drivers
		DebugDivider("!")
		Debug("COMMAND UNKNOWN (Mini Driver Switch): "..strCommand)
		DebugDivider("!")
	end
end
```

**Comment by Baubascat** [comment]
Works! Discrete on and off power.

**Comment by Greg-MM** [comment]
I just published v1.0.4, just set property On Room OFF to Turn OFF and it should work as above
